### ISSUE TITLE
Server-only minecraft jar option no longer working in Loom 1.4

### ISSUE THREAD

**Issue by LCLPYT** [FabricMC/fabric-loom#966]
Gradle builds using `loom { serverOnlyMinecraftJar() }` fail when using `fabric-loom 1.4-SNAPSHOT`.
The following error is reported by gradle:
```
Could not resolve: net.minecraft:minecraft-SERVEROnly-1f338b2c9f:1.20.2-net.fabricmc.yarn.1_20_2.1.20.2+build.1-v2
```

## Replicate the issue
- checkout [fabric-example-mod](https://github.com/FabricMC/fabric-example-mod)
- replace the loom section in the gradle buildscript with:
```diff
loom {
-    splitEnvironmentSourceSets()
-
-    mods {
-        "modid" {
-            sourceSet sourceSets.main
-            sourceSet sourceSets.client
-        }
-    }
+    serverOnlyMinecraftJar()
}
```
- sync gradle (e.g. `./gradlew build` or ide sync button)

## Error log
```
> Configure project :
Fabric Loom: 1.4.1

> Task :generateDLIConfig
> Task :generateLog4jConfig
> Task :generateRemapClasspath UP-TO-DATE
> Task :configureLaunch
> Task :ideaSyncTask
> Task :prepareKotlinBuildScriptModel UP-TO-DATE
Could not resolve: net.minecraft:minecraft-SERVEROnly-1f338b2c9f:1.20.2-net.fabricmc.yarn.1_20_2.1.20.2+build.1-v2
Could not resolve: net.minecraft:minecraft-SERVEROnly-1f338b2c9f:1.20.2-net.fabricmc.yarn.1_20_2.1.20.2+build.1-v2

BUILD SUCCESSFUL in 1s
4 actionable tasks: 3 executed, 1 up-to-date
```

**Comment by Juuxel** [comment]
I suspect it's caused by the removal of `toLowerCase` here: https://github.com/FabricMC/fabric-loom/commit/b7c80133ce9e6f13ced8ee2634f2f62690eff2e9#diff-1e4991838a7f4686fcba0a054df51f611a025eefaf3c9458f020d69b8c68add6L139-R130

**Comment by modmuss50** [comment]
I have reverted the fix for this, as it started causing the same issue for all existing projects. I think this happens becuae LocalMavenHelper is not case sensitive, but the maven resolver is.

**Comment by modmuss50** [comment]
Fixed correctly in 1.5
